priam/src/main/java/com/netflix/priam/scheduler/SchedulerType.java

```diff
@@ -0,0 +1,31 @@
+package com.netflix.priam.scheduler;
+
+/**
+ * Created by aagrawal on 3/8/17.
+ */
+public enum SchedulerType {
+    TIME("TIME"),CRON("CRON");
+
+    private final String schedulerType;
+
+    SchedulerType(String schedulerType)
+    {
+        this.schedulerType = schedulerType.toUpperCase();
+    }
+
+    public String getSchedulerType()
+    {
+        return schedulerType;
+    }
+
+    //Helper method to find the scheduler type - case insensitive as user may
+    //put value which are not right case.
+    public static SchedulerType valueOfIgnoreCase(String schedulerType) {
+        return SchedulerType.valueOf(schedulerType.toUpperCase());
+    }
+
+    public static SchedulerType lookup(String schedulerType) {
+        return valueOfIgnoreCase(schedulerType);
```

Comment:
As mentioned above, `Throwing exception might work in current scenario but might for any other future users of this enum`. You might want to check for `null` when calling this method and act accordingly (In some cases it might be a default value and in other cases, it might be throwing an exception. Adding to the current behavior, when it is fail-fast exception, it is always good idea to show the path forward in exception messages. For .e.g., Instead of throwing a message `Unsupported SchedulerType`, something like below would be helpful for OSS community.

> Unable to start Priam as `blah` is not a supported SchedulerType, supported ScheduleTypes are `HOUR` or `CRON` formats.